pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -168,7 +168,9 @@ func (configgen *ConfigGeneratorImpl) buildClusters(proxy *model.Proxy, req *mod
 		// Setup inbound clusters
 		inboundPatcher := clusterPatcher{efw: envoyFilterPatches, pctx: networking.EnvoyFilter_SIDECAR_INBOUND}
 		clusters = append(clusters, configgen.buildInboundClusters(cb, proxy, instances, inboundPatcher)...)
-		clusters = append(clusters, configgen.buildInboundHBONEClusters(proxy)...)
+		if proxy.EnableHBONE() {
+			clusters = append(clusters, configgen.buildInboundHBONEClusters())
```

Comment:
Do we need the function? We can just add MainInternalCluster here?

Reply:
Just to be more clear what does `MainInternalCluster ` do for hbone inbound